server/src/main/java/org/opensearch/indices/store/TransportNodesListShardStoreMetadataBatch.java

```diff
@@ -227,52 +239,43 @@ protected void writeNodesTo(StreamOutput out, List<NodeStoreFilesMetadataBatch>
      *
      * @opensearch.internal
      */
-    public static class NodeStoreFilesMetadata {
+    public static class NodeStoreFilesMetadata extends BaseShardResponse {
```

Comment:
why do we need the BaseShardResponse class?

Reply:
It has common functionality of providing exception faced at shard level, as well as to check if the shard level response is empty or not. Keeping this encapsulated help avoiding if/else for primary and replica in the cache code.

Removing it for now as it's not needed in this change. It'll be pushed as part of [https://github.com/opensearch-project/OpenSearch/pull/12504](https://github.com/opensearch-project/OpenSearch/pull/12504)